static/app/views/dashboards/widgetBuilder/buildSteps/groupByStep/groupBySelector.tsx

```diff
@@ -84,7 +84,7 @@ export function GroupBySelector({fieldOptions, columns = [], onChange}: Props) {
         filteredFieldOptions: FieldOptions;
       }
     );
-  }, [fieldOptions, columns]);
+  }, [fieldOptions, columnFieldsAsString]);
```

Comment:
The body of this memo doesn't use `columns` it uses `columnFieldsAsString` which is generated from `columns`.